File: 프로그래머스/lv2/12949. 행렬의 곱셈/행렬의 곱셈.js
function solution(arr1, arr2) {
    const answer = [];
    
    for (let i = 0; i < arr1.length; i++) { // arr1 행 루프
        answer.push([]);
        for (let j = 0; j < arr2[0].length; j++) { // arr2 열 루프
            let sum = 0;
            for (let k = 0; k < arr2.length; k++) { // arr2 행 루프
                sum += arr1[i][k] * arr2[k][j];
            }
            answer[i].push(sum);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):
    # 곱셈 결과 행렬 생성
    #		(3 X 2) x (2 X 2) => (3 X 2)
    ret = [[0] * len(arr2[0]) for _ in range(len(arr1))]

    # arr1의 행의 길이만큼
    for i in range(len(arr1)):
        # arr2의 열의 길이만큼
        for j in range(len(arr2[0])):
            # arr1의 열의 길이만큼
            for k in range(len(arr1[0])):
                ret[i][j] += arr1[i][k] * arr2[k][j];
    
    # 곱셈 결과 행렬 리턴
    return ret